php-frontend/src/main/java/org/sonar/php/parser/PHPGrammar.java

```diff
@@ -204,20 +205,28 @@ public NamespaceNameTree NAMESPACE_NAME() {
     return b.<NamespaceNameTree>nonterminal(PHPLexicalGrammar.NAMESPACE_NAME).is(
       b.firstOf(
         f.namespaceName(
-          b.optional(b.token(PHPPunctuator.NS_SEPARATOR)),
-          b.zeroOrMore(f.newTuple4(
-            b.firstOf(b.token(PHPLexicalGrammar.IDENTIFIER)),
-            b.token(PHPPunctuator.NS_SEPARATOR)
-          )),
-          b.token(PHPLexicalGrammar.IDENTIFIER)),
+          b.oneOrMore(f.newTuple75(
+            b.token(PHPPunctuator.NS_SEPARATOR),
+            b.token(PHPLexicalGrammar.IDENTIFIER)))),
         f.namespaceName(
-          b.token(PHPKeyword.NAMESPACE),
-          b.token(PHPPunctuator.NS_SEPARATOR),
-          b.zeroOrMore(f.newTuple6(
-            b.firstOf(b.token(PHPLexicalGrammar.IDENTIFIER)),
-            b.token(PHPPunctuator.NS_SEPARATOR)
-          )),
-          b.token(PHPLexicalGrammar.IDENTIFIER))));
+          b.firstOf(b.token(PHPLexicalGrammar.IDENTIFIER), b.token(PHPKeyword.NAMESPACE)),
+          b.zeroOrMore(f.newTuple4(
+            b.token(PHPPunctuator.NS_SEPARATOR),
+            b.token(PHPLexicalGrammar.IDENTIFIER))))));
+  }
+
+  public UseClauseTree GROUP_USE_CLAUSE() {
+    return b.<UseClauseTree>nonterminal(PHPLexicalGrammar.GROUP_USE_CLAUSE).is(
+      f.groupUseClause(
+        b.optional(
+          b.firstOf(
```

Comment:
@vilchik-elena The same firstOf() appears in USE_STATEMENT and in GROUP_USE_STATEMENT. Could we improve that?